### PR TITLE
Issue #188: Fix error with sequences

### PR DESCRIPTION
**Pull request by EzequielNicolasRafti**
EPL-668

### CODE REVIEW

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 1]
```diff
@@ -16,167 +16,204 @@
...
+      String TableName, String C_DocTypeTarget_ID, String C_DocType_ID, boolean onlyDocType,
+      boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
     }
+    final String AD_Client_ID = Utility.getContext(conn, vars, "AD_Client_ID", WindowNo);
+    final String AD_Org_ID = Utility.getContext(conn, vars, "AD_Org_ID", WindowNo);
+
+    final String cDocTypeID = (C_DocTypeTarget_ID.equals("") ? C_DocType_ID : C_DocTypeTarget_ID);
+    if (cDocTypeID.equals("")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:51]: use stringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 2]
```diff
@@ -16,167 +16,204 @@
...
+      return getDocumentNoByOrg(conn, AD_Client_ID, AD_Org_ID, TableName, updateNext);
+    }
+
+    if (AD_Client_ID.equals("0")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:55]: use stringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 3]
```diff
@@ -16,167 +16,204 @@
...
+      throw new UnsupportedOperationException("Utility.getDocumentNo - Cannot add System records");
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDocType(conn, cDocTypeID, AD_Client_ID, (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
+
+    if (cs == null || cs.razon == null || cs.razon.equals("")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:65]: use stringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 4]
```diff
@@ -16,167 +16,204 @@
...
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+    String AD_Client_ID = Utility.getContext(con, vars, "AD_Client_ID", WindowNo);
+    if ("".equals(AD_Client_ID)) {
+      AD_Client_ID = vars.getClient();
+    }
+    final String AD_Org_ID = Utility.getContext(con, vars, "AD_Org_ID", WindowNo);
+
+    final String cDocTypeID = (C_DocTypeTarget_ID.equals("") ? C_DocType_ID : C_DocTypeTarget_ID);
+    if (cDocTypeID.equals("")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:90]: use stringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 5]
```diff
@@ -16,167 +16,204 @@
...
+      return getDocumentNoByOrg(con, AD_Client_ID, AD_Org_ID, TableName, updateNext);
+    }
+
+    if (AD_Client_ID.equals("0")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:94]: use stringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 6]
```diff
@@ -16,167 +16,204 @@
...
+
+    CSResponse cs = null;
+    try {
+
+      cs = DocumentNoData.nextDocTypeConnection(conn, con, cDocTypeID, AD_Client_ID,
+          (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
+
+    if (cs == null || cs.razon == null || cs.razon.equals("")) {
```
Comment (juabaez) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:106]: use stringUtils

**Review by juabaez — CHANGES_REQUESTED** [review]

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 7]
```diff
@@ -16,167 +16,204 @@
...
+    } else {
+      return cs.razon;
+    }
+  }
+
+  @Override
+  public String getDocumentNo(Connection conn, ConnectionProvider con, VariablesSecureApp vars,
+      String WindowNo, String TableName, String C_DocTypeTarget_ID, String C_DocType_ID,
+      boolean onlyDocType, boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
```
Comment (Gremiger) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:80]: As "TableName" is a String. Is it not better to use 
 `StringUtils.isBlank(TableName) ?`

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 8]
```diff
@@ -16,167 +16,204 @@
...
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, String AD_Client_ID, String TableName,
+      boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
```
Comment (Gremiger) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:133]: Use StringUtils.isBlank()   or StringUtils.isEmpty() according to the case.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 9]
```diff
@@ -16,167 +16,204 @@
...
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDoc(conn, "DocumentNo_" + TableName, AD_Client_ID,
+          (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
```
Comment (Gremiger) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:141]: Empty catch block.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 10]
```diff
@@ -16,167 +16,204 @@
...
+   *     String with the organization id.
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  public String getDocumentNoByOrg(ConnectionProvider conn, String AD_Client_ID, String AD_Org_ID,
+      String TableName, boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
```
Comment (Gremiger) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:168]: StringUtils

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 11]
```diff
@@ -16,167 +16,204 @@
...
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDocByOrg(conn, "DocumentNo_" + TableName, AD_Client_ID, AD_Org_ID,
+          (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
```
Comment (Gremiger) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:176]: Empty catch block

**Review by Gremiger — CHANGES_REQUESTED** [review]

**Review by juabaez — APPROVED** [review]

**Review by Gremiger — APPROVED** [review]

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 12]
```diff
@@ -16,167 +17,204 @@
...
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, VariablesSecureApp vars, String WindowNo,
+      String TableName, String C_DocTypeTarget_ID, String C_DocType_ID, boolean onlyDocType,
+      boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:46]: StringUtils.isEmpty(TableName)

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 13]
```diff
@@ -16,167 +17,204 @@
...
+   * @param C_DocType_ID
+   *     id of the doctype.
+   * @param onlyDocType
+   *     Search only for doctype.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, VariablesSecureApp vars, String WindowNo,
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:42]: vars names are in mixed case with a lowercase first letter.

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 14]
```diff
@@ -16,167 +17,204 @@
...
+      String TableName, String C_DocTypeTarget_ID, String C_DocType_ID, boolean onlyDocType,
+      boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
     }
+    final String AD_Client_ID = Utility.getContext(conn, vars, "AD_Client_ID", WindowNo);
+    final String AD_Org_ID = Utility.getContext(conn, vars, "AD_Org_ID", WindowNo);
+
+    final String cDocTypeID = (C_DocTypeTarget_ID.equals("") ? C_DocType_ID : C_DocTypeTarget_ID);
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:51]: StringUtils.isEmpty(C_DocTypeTarget_ID) ? C_DocType_ID : C_DocTypeTarget_ID

or

StringUtils.defaultIfBlank(C_DocTypeTarget_ID, C_DocType_ID)

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#defaultIfBlank-T-T-

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 15]
```diff
@@ -16,167 +17,204 @@
...
+    if (StringUtils.isEmpty(cDocTypeID)) {
+      return getDocumentNoByOrg(conn, AD_Client_ID, AD_Org_ID, TableName, updateNext);
+    }
+
+    if (StringUtils.equals("0", AD_Client_ID)) {
+      throw new UnsupportedOperationException("Utility.getDocumentNo - Cannot add System records");
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:57]: Add ad_message

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 16]
```diff
@@ -16,167 +17,204 @@
...
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDocType(conn, cDocTypeID, AD_Client_ID, (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:64]: Noting to do?

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 17]
```diff
@@ -16,167 +17,204 @@
...
+      return cs.razon;
+    }
+  }
+
+  @Override
+  public String getDocumentNo(Connection conn, ConnectionProvider con, VariablesSecureApp vars,
+      String WindowNo, String TableName, String C_DocTypeTarget_ID, String C_DocType_ID,
+      boolean onlyDocType, boolean updateNext) {
+    if (StringUtils.isEmpty(TableName)) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:82]: Add ad_message

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 18]
```diff
@@ -16,167 +17,204 @@
...
+      } else {
+        return "0";
+      }
+    } else {
+      return cs.razon;
+    }
+  }
+
+  @Override
+  public String getDocumentNo(Connection conn, ConnectionProvider con, VariablesSecureApp vars,
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:78]: vars names are in mixed case with a lowercase first letter.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 19]
```diff
@@ -16,167 +17,204 @@
...
+    if (StringUtils.isEmpty(TableName)) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+    String AD_Client_ID = Utility.getContext(con, vars, "AD_Client_ID", WindowNo);
+    if (StringUtils.isBlank(AD_Client_ID)) {
+      AD_Client_ID = vars.getClient();
+    }
+    final String AD_Org_ID = Utility.getContext(con, vars, "AD_Org_ID", WindowNo);
+
+    final String cDocTypeID = (C_DocTypeTarget_ID.equals("") ? C_DocType_ID : C_DocTypeTarget_ID);
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:90]: You can also do it as:

String cDocTypeID = StringUtils.defaultIfBlank(C_DocTypeTarget_ID, C_DocType_ID);

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 20]
```diff
@@ -16,167 +17,204 @@
...
+      String WindowNo, String TableName, String C_DocTypeTarget_ID, String C_DocType_ID,
+      boolean onlyDocType, boolean updateNext) {
+    if (StringUtils.isEmpty(TableName)) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+    String AD_Client_ID = Utility.getContext(con, vars, "AD_Client_ID", WindowNo);
+    if (StringUtils.isBlank(AD_Client_ID)) {
+      AD_Client_ID = vars.getClient();
+    }
+    final String AD_Org_ID = Utility.getContext(con, vars, "AD_Org_ID", WindowNo);
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:88]: vars names are in mixed case with a lowercase first letter.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 21]
```diff
@@ -16,167 +17,204 @@
...
+      throw new UnsupportedOperationException("Utility.getDocumentNo - Cannot add System records");
+    }
+
+    CSResponse cs = null;
+    try {
+
+      cs = DocumentNoData.nextDocTypeConnection(conn, con, cDocTypeID, AD_Client_ID,
+          (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:105]: Noting to do?

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 22]
```diff
@@ -16,167 +17,204 @@
...
+   * @param AD_Client_ID
+   *     String with the client id.
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, String AD_Client_ID, String TableName,
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:132]: vars names are in mixed case with a lowercase first letter.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 23]
```diff
@@ -16,167 +17,204 @@
...
+   *     String with the client id.
+   * @param AD_Org_ID
+   *     String with the organization id.
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  public String getDocumentNoByOrg(ConnectionProvider conn, String AD_Client_ID, String AD_Org_ID,
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:167]: vars names are in mixed case with a lowercase first letter.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 24]
```diff
@@ -16,167 +17,204 @@
...
+      String TableName, boolean updateNext) {
+    if (StringUtils.isEmpty(TableName)) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:170]: Add ad_message

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 25]
```diff
@@ -16,167 +17,204 @@
...
+   *     String with the client id.
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNoConnection(Connection conn, ConnectionProvider con,
+      String AD_Client_ID, String TableName, boolean updateNext) {
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:202]: vars names are in mixed case with a lowercase first letter.

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 26]
```diff
@@ -16,167 +17,204 @@
...
+    if (TableName == null || TableName.length() == 0) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:204]: Add ad_message

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 27]
```diff
@@ -16,167 +17,204 @@
...
+   * @param TableName
+   *     Table name.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNoConnection(Connection conn, ConnectionProvider con,
+      String AD_Client_ID, String TableName, boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:203]: StringUtils.isEmpty(TableName)

File: modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java [hunk 28]
```diff
@@ -16,167 +17,204 @@
...
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDocConnection(conn, con, "DocumentNo_" + TableName, AD_Client_ID,
+          (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
```
Comment (Matias-Bernal) [inline modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java:212]: Noting to do?

File: src/org/openbravo/erpCommon/utility/DocumentNo_data.xsql [hunk 29]
```diff
@@ -49,6 +49,17 @@
       <Parameter name="updateNext"/>
       <Parameter name="razon" type="out"/>
    </SqlMethod>
+   <SqlMethod name="nextDocByOrg" type="callableStatement" return="object" object="CSResponse">
+      <SqlMethodComment></SqlMethodComment>
+      <Sql><![CDATA[
+        CALL legap_sequence_doc_by_org(?,?,?,?,?)
```
Comment (Matias-Bernal) [inline src/org/openbravo/erpCommon/utility/DocumentNo_data.xsql:55]: do you check it on Oracle Database?

**Review by Matias-Bernal — CHANGES_REQUESTED** [review]

**Review by Gremiger — CHANGES_REQUESTED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Pull request closed without merging**